libs/portals/admin/service-desk/src/screens/Users/Users.action.ts

```diff
@@ -0,0 +1,108 @@
+import { z } from 'zod'
+import { redirect } from 'react-router-dom'
+import { isEmail } from 'class-validator'
+import * as kennitala from 'kennitala'
+import { parsePhoneNumber } from 'libphonenumber-js'
+
+import {
+  RawRouterActionResponse,
+  WrappedActionFn,
+} from '@island.is/portals/core'
+import {
+  replaceParams,
+  validateFormData,
+  ValidateFormDataResult,
+} from '@island.is/react-spa/shared'
+import { isSearchTermValid, maskString } from '@island.is/shared/utils'
+
+import {
+  GetPaginatedUserProfilesDocument,
+  GetPaginatedUserProfilesQuery,
+  type GetPaginatedUserProfilesQueryVariables,
+} from './Users.generated'
+import { ServiceDeskPaths } from '../../lib/paths'
+
+export enum ErrorType {
+  // Add more error types here when needed
+  InvalidSearchQuery = 'INVALID_SEARCH_QUERY',
+}
+
+const schema = z.object({
+  searchQuery: z.string().min(1),
```

Comment:
You can do the `isSearchTermValid` check here be using `refine`
https://zod.dev/?id=refine